Sources/Client/Utils/NestableCodingKey.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  NestableCodingKey.swift
+//  StreamChatClient
+//
+//  Created by Bahadir Oncel on 1.04.2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+// Courtesy of AliSoftware
+// https://gist.github.com/AliSoftware/89b275d7259d23ebf12d377b6ffe15cd
+
+import Foundation
+
+/// Use this to annotate the properties that require a depth traversal during decoding.
+/// The corresponding `CodingKey` for this property must be a `NestableCodingKey`
+@propertyWrapper
+public struct NestedKey<T: Decodable>: Decodable {
+    struct AnyCodingKey: CodingKey {
+        let stringValue: String
+        let intValue: Int?
+        init(stringValue: String) {
```

Comment:
Add an empty before init. Please check the style if you copy/paste code from somewhere.